src/main/scala/tech/cryptonomic/conseil/routes/PlatformDiscovery.scala

```diff
@@ -21,36 +27,55 @@ object PlatformDiscovery {
   * @param config              configuration object
   * @param apiExecutionContext is used to call the async operations exposed by the api service
   */
-class PlatformDiscovery(config: PlatformsConfiguration)(implicit apiExecutionContext: ExecutionContext) extends LazyLogging with RouteHandling {
+class PlatformDiscovery(config: PlatformsConfiguration)(implicit apiExecutionContext: ExecutionContext, system: ActorSystem) extends LazyLogging with RouteHandling {
+
+  /** default caching settings*/
+  private val defaultCachingSettings: CachingSettings = CachingSettings(system)
```

Comment:
Instead of passing the `system` to indirectly read the caching config, we should first validate the configuration in a wrapping pureconfig object and then pass that down here to use with the `Config` based `CachingSettings.apply`.

For reference, you can look into how HostConnectionPooling configuration is passed [here](https://github.com/Cryptonomic/Conseil/blob/4e6f32bbcab0289542d46d4a795e540e9eddb03a/src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeInterface.scala#L181), and how it's initially validated

Reply:
I'm not sure about that, `CachingSettings.apply` takes config from whole path `akka.http.caching.lfu-cache`, so I would have to keep whole config there

Reply:
if you look at the whole process of how the host-pool is defined, you'll find that you can read a "sub-config" from anywhere and then put it at a given path, such as `akka.http.caching.lfu-cache`, as expected by akka.

Reply:
I've refactored a bit reading config to be more generic, as I could reuse it